lib/zinnia.js

```diff
@@ -49,13 +44,33 @@ const maybeReportCrashToSentry = (/** @type {unknown} */ err) => {
   Sentry.captureException(err)
 }
 
+const updateAllSourceFiles = async () => {
+  const modules = await Promise.all(
+    Object
+      .values(ZINNIA_MODULES)
+      .map(({ module, repo }) =>
+        pRetry(
+          () => updateSourceFiles({ module, repo }),
```

Comment:
This can't happen. If one module fails to update, we have a systemic problem and the whole process will go down.

Reply:
Sorry if I was not clear. This is not about the "cannot update" error case, it can happen on the happy path when restarting Station.

1. Station runs with the latest version of modules.
2. The user restarts the Station.
3. While Station is starting (and Zinnia is loading modules), the auto-updater detects a new module version
4. Zinnia starts and loads some but not all module source files yet
5. Update check detects a new version and starts downloading & unpacking new source code
6. As Zinnia loads more module source files, it will eventually start reading files from the new version.

I admit the time window for this race condition to happen is very small: how long does it take Zinnia to read a JS file from the disk, parse it and discover ESM imports to load? But as [the saying goes](https://en.wikipedia.org/wiki/Murphy's_law): _Anything that can go wrong will go wrong, and at the worst possible time._

Reply:
> While Station is starting (and Zinnia is loading modules), the auto-updater detects a new module version

This can't happen though. We _always_ update to the latest module versions on start, and then only check for updates 10 minutes later (and every following 10 minutes), after having successfully updated.

Reply:
Got it!

Is it worth adding a code comment to point out this guarantee to future readers?

Does the same argument apply when we restart Zinnia after a new module version is downloaded? I.e. we will check for more updates 10 minutes after the last restart?

Are we running the update logic twice - the first time as part of the periodic background check for new module versions and the second time as part of the Zinnia restart?

Maybe this is telling us that we can simplify the periodic update checks so that they don't install the new version; just restart the entire "run Zinnia modules" function so that the function downloads the new module versions first and then starts Zinnia. The downside is a bit longer downtime, but I would hope it shouldn't be significant.

Anyhow, it's just an idea to consider.

If you feel the current implementation is robust enough, then I am fine with it too.